pkg/http/api_test.go

```diff
@@ -137,14 +137,14 @@ func TestPubSubEndpoints(t *testing.T) {
 		}
 	})
 
-	t.Run("Publish without topic name ending in // - 404", func(t *testing.T) {
+	t.Run("Publish with topic name '/' - 204", func(t *testing.T) {
```

Comment:
We had a PR to decode the URL, so the client request actually ends with "%2F" and not "/".

Can you try running daprd and invoking the following endpoints? Not in unit-tests, the actual result in daprd.

http://localhost:3500/publish/pubsubname//
http://localhost:3500/publish/pubsubname/%2F
http://localhost:3500/publish/pubsubname

Reply:
Yes, i see.

Because [fasthttp/router pr47](https://github.com/fasthttp/router/pull/47) was merged,  UT case:  "Publish without topic name ending in // - 404" can't pass.


http://localhost:3500/publish/pubsubname// , this request param `topicParam`'s value is /, not empty string. If i modify topic == "/" to topic == "", dapr runtime can't enter error logic, and return status code:404 

```golang
// github.com/dapr/dapr/pkg/http/api.go: onPublish
topic := reqCtx.UserValue(topicParam).(string)
if topic == "/" {
    msg := NewErrorResponse("ERR_TOPIC_EMPTY", fmt.Sprintf(messages.ErrTopicEmpty,   pubsubName))
    respondWithError(reqCtx, fasthttp.StatusNotFound, msg)
    log.Debug(msg)
    return
}
```

Reply:
http://localhost:3500/publish/pubsubname// maps to topicName as "/" and results in 204
http://localhost:3500/publish/pubsubname/%2F also maps to topicName as "/" and results in 204
http://localhost:3500/publish/pubsubname maps to topicName as "" and results in 404